src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,262 @@
+# WRoverPlayground - Topic Training
+Now that you have completed the Tools Training, you are ready to learn about ROS topics. From the Setup Training, you should be familiar with:
+- the basics of ROS Nodes
+- `rosnode`
+- `roscore`
+- `rqt_graph`
+- `rqt_plot`.
+
+Once you complete this training, you should be familiar with:
+- the purpose of topics
+- the relationship between Publishers and Subscribers
+- setting up Publishers and Subscribers
+- running Publishers and Subscribers
+- custom topic types
+- and `rostopic type {topic}`.
+
+## The Purpose of Topics
+Though possible, it is not feasible to have one node contain all the information a rover would need. We need to spread out the information into different files. This is where ROS topics come in. The word "topic" sounds very general, but it is actually a specific concept that allows ROS nodes to communicate with each other. They are a way to stream data in and out of a program.
+
+Using topics, our code now becomes 'hot-swappable.' Topics do not care which node is sending or receiving data, so we can easily create a node that serves data from a physical sensor as well as a mock node to test other nodes without needing to change those nodes. The reason is because the interfaces provided through the published data versus an actual sensor are equivalent.
+
+This allows us to specialize nodes, so they can execute their own code/tasks, while also sending and receiving information. For example, a node could receive data and send it to another one, a node could receive and analyze data and have another node do some action with it, a node could send data to multiple nodes, etc.
```

Comment:
Maybe give a specific example, to make it concrete?

Reply:
@bennowotny Should this just be a real-world example, like USBs?

Reply:
Maybe something we use on the robot, like the specialization of the motor driver programs being separated from how we drive the motors which is separated from how we gather user input.

Reply:
Fixed in f8df283.